### PR TITLE
fix(Tooltip): fix useLayoutEffect ssr warning

### PR DESCRIPTION
**Pull request by xendke**
### Overview

We are using a lot of floating tooltips in a new portal app page, but SSR nextjs throws a lot of warnings having to do with the use of `useLayoutEffect` inside FloatingTooltip.

> ⚠️ Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer’s output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://fb.me/react-uselayouteffect-ssr for common fixes.

- This PR is a small update to prevent FloatingTooltip to render in SSR. Followed React's suggestions to lazily show the component. Which in our case works pretty well since we can fallback to the non-floating tooltip on SSR.

The other [suggestion](https://fb.me/react-uselayouteffect-ssr) is to just use `useEffect` but I assume there is a reason we're not.

### PR Checklist

- [x] Related to designs: None
- [x] Related to JIRA ticket: None
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

Tooltip should continue to behave as it was previously. 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.183-alpha.da947b.0<br/>@codecademy/gamut-labs@33.1.4-alpha.da947b.0<br/>@codecademy/gamut@50.4.2-alpha.da947b.0<br/>@codecademy/monaco@0.6.75-alpha.da947b.0<br/>@codecademy/styleguide@61.2.2-alpha.da947b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62ffea0e73d245367e1fa3f3--gamut-preview.netlify.app>https://62ffea0e73d245367e1fa3f3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62ffea0e73d245367e1fa3f3>Deploy Logs</a></p>

File: packages/gamut/src/ToolTip/ToolTip.tsx [hunk 1]
```diff
@@ -9,9 +9,16 @@ export const ToolTip: React.FC<ToolTipProps> = ({
...
+  useEffect(() => {
+    setLoaded(true);
+  }, []);
+
+  return placement === 'floating' && loaded ? (
     <FloatingToolTip alignment={alignment} widthMode={widthMode} {...rest} />
   ) : (
     <InlineToolTip alignment={alignment} widthMode={widthMode} {...rest} />
   );
+};
```
Comment (jakemhiller) [inline packages/gamut/src/ToolTip/ToolTip.tsx:24]: is there a reason not to exclude both from SSR? since they don't need to show up immediately?

**Review by jakemhiller — CHANGES_REQUESTED** [review]

File: packages/gamut/src/ToolTip/ToolTip.tsx (hunk 1 above)
Comment (xendke) [reply packages/gamut/src/ToolTip/ToolTip.tsx:24]: The ToolTip component renders the `target` component inside. so preventing them from SSR would mean these little icons would not SSR.

Screen Shot 2022-08-22 at 11 44 24 AM

Comment (jakemhiller) [reply packages/gamut/src/ToolTip/ToolTip.tsx:24]: hm, fair enough. i feel like there's a better solution here that doesn't involve re-rendering the target on load, but this should work for now

**Review by jakemhiller — APPROVED** [review]

**Pull request merged**